react/features/calendar-sync/web/microsoftCalendar.js

```diff
@@ -0,0 +1,492 @@
+
+/**
+ * The URL to use when authenticating using Microsoft API.
+ * @type {string}
+ */
+const AUTH_ENDPOINT
+    = 'https://login.microsoftonline.com/common/oauth2/v2.0/authorize?';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * Loads and interacts with the Microsoft Calendar API.
+ */
+export class MicrosoftCalendarApi {
+    /**
+     * The ID for the Microsoft client application used to init the API.
+     */
+    _appClientID: string;
+
+    /**
+     * The current request data.
+     */
+    _currentRequest: ?Object;
+
+    /**
+     * The last valid response data.
+     */
+    _lastValidResponse: ?Object;
+
+    /**
+     * The popup window used for authentication, using the instance to close
+     * the popup once we have all the info we need.
+     */
+    _popupAuthWindow: ?Object;
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    _dispatch: Dispatch<*>;
+
+    /**
+     * The redux function that gets/retrieves the redux state.
+     */
+    _getState: Function;
+
+    /**
+     * Initializes a new Microsoft Calendar API instance.
+     *
+     * @param {string} appClientID - The ID for the Microsoft client application
+     * used to init the API.
+     * @param {Object} store - The redux store.
+     */
+    constructor(appClientID: string, store: Object) {
+        this._appClientID = appClientID;
+        this._dispatch = store.dispatch;
+        this._getState = store.getState;
+        this._currentRequest = {};
+
+        // checks post messages for events from the popup
+        window.addEventListener('message', this._onPostMessage.bind(this));
+    }
+
+    /**
+     * Initializes the api.
+     *
+     * @returns {function(Dispatch<*>): Promise<void>}
+     */
+    init(): Promise<void> {
+        this._dispatch(setCalendarAPIState(CALENDAR_API_STATES.LOADED));
+
+        return Promise.resolve();
+    }
+
+    /**
+     * Retrieves the current calendar events.
+     *
+     * @param {number} fetchStartDays - The number of days to go back
+     * when fetching.
+     * @param {number} fetchEndDays - The number of days to fetch.
+     * @returns {function(Dispatch<*>): Promise<CalendarEntries>}
+     */
+    getCalendarEntries(
+            fetchStartDays: ?number, fetchEndDays: ?number) {
+        return (dispatch: Dispatch<*>, getState: Function): Promise<*> => {
+
+            // not authorized, skip
+            if (!getState()['features/calendar-sync'].msAuthState) {
+                return Promise.resolve();
+            }
+
+            return new Promise((resolve, reject) => {
+                const client = Client.init({
+                    authProvider: done => done(
+                        null,
+                        getState()['features/calendar-sync']
+                            .msAuthState.accessToken)
+                });
+
+                const query = client.api('/me/calendars');
+
+                query.get((error, response) => {
```

Comment:
I read this in the microsoft-graph-client readme: `The actions(.get(), .put(), etc.) accept a callback or don't pass in a function to get back a Promise.` That might mean you don't need the wrapper.